treasury-factory/src/lib.rs

```diff
@@ -56,17 +60,95 @@ impl Contract {
                 Gas::from_tgas(80),
             )
             .then(
-                instance_contract::ext(new_instance_contract_id.clone())
-                    .with_attached_deposit(NearToken::from_near(1))
-                    .update_widgets(widget_reference_account_id, social_db_account_id),
-            )
-            .then(
-                sputnik_dao::ext(sputnik_dao_factory_account_id.parse().unwrap())
-                    .with_attached_deposit(NearToken::from_near(6))
-                    .with_static_gas(Gas::from_tgas(100))
-                    .create(name.to_string(), create_dao_args),
+                Self::ext(env::current_account_id()).create_account_callback(
+                    env::predecessor_account_id(),
+                    env::attached_deposit(),
+                    name,
+                    new_instance_contract_id,
+                    sputnik_dao_factory_account_id,
+                    social_db_account_id,
+                    widget_reference_account_id,
+                    create_dao_args,
+                ),
             )
     }
+
+    pub fn create_account_callback(
+        &self,
+        refund_on_failure_account: AccountId,
+        refund_on_failure_amount: NearToken,
+        name: String,
+        new_instance_contract_id: AccountId,
+        sputnik_dao_factory_account_id: String,
+        social_db_account_id: String,
+        widget_reference_account_id: String,
+        create_dao_args: String,
+    ) -> Promise {
+        if env::predecessor_account_id() != env::current_account_id() {
+            env::panic_str("Should not be called directly");
+        }
```

Comment:
You can just add [`#[private]` attribute](https://docs.rs/near-sdk/latest/near_sdk/attr.near.html#private-annotates-methods-of-a-type-in-its-impl-block) to the function instead.